src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/model/SentenceCalculation.kt

```diff
@@ -266,29 +310,90 @@ data class SentenceCalculation(
+      ConsecutiveSentenceAggregator((sentence as ConsecutiveSentence).orderedSentences.map { it.custodialDuration() }).calculateDays(
+        sentence.sentencedAt,
+      )
+    } else {
+      val custodialDuration = sentence.custodialDuration()
+      custodialDuration.getLengthInDays(sentence.sentencedAt)
+    }
+
+    val effectiveRelease = extendedDeterminateParoleEligibilityDate ?: adjustedDeterminateReleaseDate
+    val maxEffectiveErsed = effectiveRelease.minusDays(MAX_ERSED_PERIOD_DAYS.toLong())
+    val maxEffectiveErsedReleaseCalcBreakdown = ReleaseDateCalculationBreakdown(
+      rules = setOf(CalculationRule.ERSED_MAX_PERIOD),
+      releaseDate = maxEffectiveErsed,
+      unadjustedDate = effectiveRelease,
+      adjustedDays = ChronoUnit.DAYS.between(
+        unadjustedDeterminateReleaseDate,
+        adjustedDeterminateReleaseDate,
+      ).toInt(),
+      rulesWithExtraAdjustments = mapOf(
+        CalculationRule.ERSED_MAX_PERIOD to AdjustmentDuration(
+          -MAX_ERSED_PERIOD_DAYS,
+          ChronoUnit.DAYS,
+        ),
+      ),
+    )
+    val release = unadjustedExtendedDeterminateParoleEligibilityDate ?: unadjustedDeterminateReleaseDate
+
+    val daysUntilRelease = ChronoUnit.DAYS.between(sentence.sentencedAt, release).plus(1).toInt()
+    val unadjustedErsed =
+      sentence.sentencedAt
+        .plusDays(ceil(daysUntilRelease.toDouble() / 2).toLong())
+    val minimumEffectiveErsed = unadjustedErsed
+      .plusDays(calculatedTotalAddedDays.toLong())
+      .minusDays(calculatedTotalDeductedDays.toLong())
+      .plusDays(calculatedTotalAwardedDays.toLong())
+    val minimumEffectiveErsedReleaseCalcBreakdown = ReleaseDateCalculationBreakdown(
+      rules = setOf(CalculationRule.ERSED_MIXED_TERMS),
+      releaseDate = minimumEffectiveErsed,
+      unadjustedDate = unadjustedErsed,
+      adjustedDays = ChronoUnit.DAYS.between(unadjustedErsed, minimumEffectiveErsed).toInt(),
+    )
+
+    return if (minimumEffectiveErsed.isAfter(maxEffectiveErsed)) {
+      minimumEffectiveErsedReleaseCalcBreakdown
+    } else {
+      maxEffectiveErsedReleaseCalcBreakdown
+    }
+  }
+
   private fun calculateErsedFromTwoThirds(): ReleaseDateCalculationBreakdown {
     val days = if (sentence is ConsecutiveSentence) {
-      ConsecutiveSentenceAggregator((sentence as ConsecutiveSentence).orderedSentences.map { it.custodialDuration() }).calculateDays(sentence.sentencedAt)
+      ConsecutiveSentenceAggregator((sentence as ConsecutiveSentence).orderedSentences.map { it.custodialDuration() }).calculateDays(
+        sentence.sentencedAt,
+      )
     } else {
       val custodialDuration = sentence.custodialDuration()
       custodialDuration.getLengthInDays(sentence.sentencedAt)
     }
     return if (days >= RELEASE_AT_TWO_THIRDS_ERSED_DAYS) {
       val release = extendedDeterminateParoleEligibilityDate ?: releaseDate
-      val ersed = release.minusYears(1)
+      val ersed = release.minusDays(MAX_ERSED_PERIOD_DAYS.toLong())
```

Comment:
nice work moving these hard coded numbers to constants - makes it easier to read

Reply:
🙌